my_game.py

```diff
@@ -144,6 +243,16 @@ def __init__(self, width, height):
         # Set the background color
         arcade.set_background_color(arcade.color.AMAZON)
 
+    def spawn_ufo(self, delta_time):
+        """
+        spawns an ufo object into self.ufo_list.
+        has to take delta_time because it needs to be called by arcade.schedule
+        """
+
+        new_ufo_obj = BonusUFO()
+        new_ufo_obj.__int__(self.ufo_shot_list)  # it needs the list so it can send shots to MyGame
+        self.ufo_list.append(new_ufo_obj)
```

Comment:
Why not:
```
self.ufo_list.append(BonusUFO(self.ufo_shot_list))
```

Reply:
Instances of classes created in a scheduled function don't __init__. It has to be maually called. So it needs to be this way

Reply:
What happens if you do this?:
```
new_ufo_obj = BonusUFO(self.ufo_shot_list)
self.ufo_list.append(new_ufo_obj)
```

It seems wrong that you can create a BonusUFO without passing a SpriteList.

Reply:
thats true... i don know why it works.
Maybe because it doesn't run __init__ it doesn't ask for a shot list?

Reply:
I'll allow it. We can talk about those issues and try to figure them out later :+1: